modules/odata/odata-core/src/main/java/org/eclipse/dirigible/engine/odata2/sql/processor/AbstractSQLProcessor.java

```diff
@@ -634,25 +657,70 @@ public BatchResponsePart doExecuteChangeSet(final BatchHandler handler, final Li
         context.put(SQL_CONTEXT_CONTEXT_KEY, createSQLContext(connection));
         context.put(DATASOURCE_CONTEXT_KEY, getDataSource());
+        context.put(ODATA_CONTEXT_CONTEXT_KEY, getContext());
     }
 
-    private void updateDeleteEventHandlerContext(Map<Object, Object> context, DeleteUriInfo uriInfo, Connection connection) throws ODataException, SQLException {
+    private void updateDeleteEventHandlerContext(Map<Object, Object> context, DeleteUriInfo uriInfo, Map<String, Object> keys, Connection connection, EdmEntitySet entitySet) throws ODataException, SQLException {
+        context.put(DUMMY_BUILDER_CONTEXT_KEY, this.getSQLQueryBuilder().buildDeleteEntityQuery((UriInfo) uriInfo, keys, getContext()));
         context.put(SELECT_BUILDER_CONTEXT_KEY, this.getSQLQueryBuilder().buildSelectEntityQuery((UriInfo) uriInfo, getContext()));
         context.put(SQL_CONTEXT_CONTEXT_KEY, createSQLContext(connection));
         context.put(DATASOURCE_CONTEXT_KEY, getDataSource());
+
+        putEntryInsertBuilderInContext(context, BEFORE_DELETE_ENTRY_JSON, (UriInfo) uriInfo, entitySet);
     }
 
-    private void updateUpdateEventHandlerContext(Map<Object, Object> context, Connection connection, PutMergePatchUriInfo uriInfo, ODataEntry entry) throws ODataException, SQLException {
+    private void updateUpdateEventHandlerContext(Map<Object, Object> context, Connection connection, PutMergePatchUriInfo uriInfo, ODataEntry entry, EdmEntitySet entitySet) throws ODataException, SQLException {
+        context.put(DUMMY_BUILDER_CONTEXT_KEY, this.getSQLQueryBuilder().buildUpdateEntityQuery((UriInfo) uriInfo, entry,
+                mapKeys(uriInfo.getKeyPredicates()), getContext()));
         context.put(SELECT_BUILDER_CONTEXT_KEY, this.getSQLQueryBuilder().buildSelectEntityQuery((UriInfo) uriInfo, getContext()));
         context.put(UPDATE_BUILDER_CONTEXT_KEY, this.getSQLQueryBuilder().buildUpdateEntityQuery((UriInfo) uriInfo, entry,
                 mapKeys(uriInfo.getKeyPredicates()), getContext()));
         context.put(SQL_CONTEXT_CONTEXT_KEY, createSQLContext(connection));
         context.put(DATASOURCE_CONTEXT_KEY, getDataSource());
+
+        putEntryInsertBuilderInContext(context, BEFORE_UPDATE_ENTRY_JSON, (UriInfo) uriInfo, entitySet);
+    }
+
+    private void putEntryInsertBuilderInContext(Map<Object, Object> context, String key, UriInfo uriInfo, EdmEntitySet entitySet) throws ODataException {
+        if (context.containsKey(key)) {
+            String beforeUpdateEntryJSON = (String) context.get(key);
+            InputStream entryContent = new ByteArrayInputStream(beforeUpdateEntryJSON.getBytes(StandardCharsets.UTF_8));
+            ODataEntry parsedEntry = parseEntry(entitySet, entryContent, ContentType.APPLICATION_JSON.toContentTypeString(), false);
+            context.put(INSERT_BUILDER_CONTEXT_KEY, this.getSQLQueryBuilder().buildInsertEntityQuery(uriInfo, parsedEntry, getContext()));
+        }
     }

```

Comment:
Is it possible to put in the context - uriInfo, entry, getDataSource and ODataContext everything else could be build based on them. This way you will not need to create and add sql queries specific for the ScriptingHandler implementation and each implementation will be able to use them to implement it's logic.

Also the input stream should be closed.